### DISCUSSION TITLE
GraphNode needs a way to Collapse and Expand

### DISCUSSION THREAD

**Discussion by donn-xx** [godotengine/godot-proposals#8280] [Editor]
A graph node should be able to collapse *and* take the noodles with it; and then expand again on-command.

The noodles could all point to one place (adjustable):
![image](https://github.com/godotengine/godot-proposals/assets/392218/74a1d2bb-6e41-4f37-8573-8c8fe676eee3)

Or they could collapse into dots that surround the node (as Blender does it)
![image](https://github.com/godotengine/godot-proposals/assets/392218/b3eff662-ed40-43ce-803a-68b92af6b0a5)

I think this is simply vital. 

Thoughts?

**Comment by GeminiSquishGames** [comment]
Yes, do want much! I think most node-based things work like this. Blender and ComfyUI off the top of my head. That and compactable Node Groups feel like the equivalent of IDE's ability to fold blocks of code, IMO. Makes it easy on the eyes and organization is key with visual or textual logic.

**Comment by Calinou** [comment]
GraphNode frames are supported since 4.3, but they can't be collapsed so far: https://github.com/godotengine/godot/pull/88014

**Reply by donn-xx** [reply]
Is the OP what a Graph Frame is? It seems rather to be a kind of grouping of multiple nodes.
